--- a/CMakeLists.txt	2019-07-28 12:55:02.852946192 +0200
+++ b/CMakeLists.txt	2019-07-28 12:57:15.570034794 +0200
@@ -591,7 +591,12 @@
 endif(USE_STATIC_BOOST)
 
 find_package(Boost REQUIRED COMPONENTS thread system)
-target_link_libraries(domoticz Boost::thread Boost::system)
+
+## Fix ld looking for the .so in the "-boost-1.70.0-dev" path
+target_link_libraries(domoticz "boost_thread" "boost_system")
+
+## Dynamically link python3 to avoid domoticz dlopen-ing at start
+target_link_libraries(domoticz "python3")
 
 # compare found vs required libBoost version
 if(Boost_VERSION VERSION_LESS DOMO_MIN_LIBBOOST_VERSION)
